src/const_eval.md

```diff
@@ -67,6 +67,30 @@ surrounding generic parameters: such an expression must either be a single bare
 const generic parameter, or an arbitrary expression not making use of any
 generics.
 
+## Permitted use of `static`s
+
+Within the constant evaluation context, items that ultimately contain no mutable memory or data that is
+capable of interior mutability can be used, borrowed or taken address of.
+Among those are the `static` items as long as they are not actually `mut` items.
+
+```rust
+const fn allowed() -> &'static u32 {
+    static VALUE: u32 = 0;
+    &VALUE
+}
+const A_CONST: &'static u32 = allowed();
+```
+
+On the contrary, for example, `static mut` and types embedding an `UnsafeCell` is not allowed.
+
+```rust,edition2021,compile_fail
```

Comment:
`compile_fail` should usually have an E code to ensure the correct error is being emitted.

```suggestion
```rust,edition2021,compile_fail,E0080
```